File: veriSaklamak/app/src/main/java/com/mhmmdyzci/verisaklamak/MainActivity.java
package com.mhmmdyzci.verisaklamak;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    EditText editText;
    TextView textView;
    SharedPreferences sharedPreferences;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        // bu veriyi telefonun kendi hafızasıda saklar
        sharedPreferences = this.getSharedPreferences("com.mhmmdyzci.verisaklamak", Context.MODE_PRIVATE);
        editText = findViewById(R.id.editTextNumber);
        textView = findViewById(R.id.textView);

        int storedAge = sharedPreferences.getInt("storedAge", 0);
        if (storedAge == 0){
            textView.setText("Your Age: ");
        }else {
        textView.setText("Your Age: " + storedAge);}

    }
    public void save(View view){
        if(!editText.getText().toString().matches(" ")){
            int age = Integer.parseInt(editText.getText().toString());
            textView.setText("Your age: " + age);
            sharedPreferences.edit().putInt("storedAge",age).apply();
        }

    }
    public  void delete(View view){
        int storedData = sharedPreferences.getInt("storedAge", 0);
        if( storedData != 0){
            sharedPreferences.edit().remove("storedAge").apply();
            textView.setText("Your Age: ");
        }
    }
}

File: StoringDataKotlin/app/src/main/java/com/mhmmdyzci/storingdatakotlin/MainActivity.kt
package com.mhmmdyzci.storingdatakotlin
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.mhmmdyzci.storingdatakotlin.databinding.ActivityMainBinding
class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private lateinit var sharedPreferences : SharedPreferences
     var myAge : Int? = null
     var ageFromDatabase : Int? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
        sharedPreferences = this.getSharedPreferences("com.mhmmdyzci.storingdatakotlin", MODE_PRIVATE)
        ageFromDatabase = sharedPreferences.getInt("age",-1)
        if (ageFromDatabase == -1  ){
            binding.textView.text = "Your age: "
        }else{
            binding.textView.text = "Your age: " + ageFromDatabase
        }

    }

    fun save(view: View){
        //SharedPreferences
        myAge = binding.textView.toString().toIntOrNull()
        if (myAge != null){
            binding.textView.text = "Your age: " + myAge
            sharedPreferences.edit().putInt("age",myAge!!).apply()

        }else{
            binding.textView.text = "Error"
        }
    }
    fun delete(view:View){
        ageFromDatabase = sharedPreferences.getInt("age",-1)
        if (ageFromDatabase !=-1 ){
            sharedPreferences.edit().remove("age").apply()
            binding.textView.text = "Your age: "

        }


    }
}